### PR TITLE
Upgrade all kinds of dependencies & use kork-bom

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,24 +5,18 @@ buildscript {
...
-    classpath "com.netflix.nebula:nebula-kotlin-plugin:$kotlinVersion"
-    classpath "com.netflix.spinnaker.gradle:spinnaker-gradle-project:4.0.0"
-    classpath "org.jetbrains.dokka:dokka-gradle-plugin:0.9.15"
-    classpath "org.jetbrains.kotlin:kotlin-allopen:$kotlinVersion"
-  }
-  configurations.all {
-    resolutionStrategy {
-      eachDependency { details ->
-        if (details.requested.group == "org.jetbrains.kotlin") {
-          details.useVersion kotlinVersion
```
Comment (robfletcher) [inline build.gradle]: No longer need this as we get `kotlin-bom` transitively from `kork-bom` so that aligns everything.

File: build.gradle [hunk 2]
```diff
@@ -36,22 +30,19 @@ subprojects {
   }
 
   dependencies {
-    compile "org.slf4j:slf4j-api:1.7.25"
-    testRuntime "org.slf4j:slf4j-nop:1.7.25"
-  }
-
-  spinnaker {
-    dependenciesVersion = "0.127.0"
```
Comment (robfletcher) [inline build.gradle]: No longer need spinnaker-dependencies

File: build.gradle [hunk 3]
```diff
@@ -36,22 +30,19 @@ subprojects {
...
+    api(platform("com.netflix.spinnaker.kork:kork-bom:4.0.0-rc.7-bom"))
```
Comment (robfletcher) [inline build.gradle]: Everything comes from this

File: gradle.properties [hunk 4]
```diff
@@ -1,4 +1 @@
 org.gradle.parallel=true
-kotlinVersion=1.2.61
-junitVersion=1.2.0
-spekVersion=1.1.5
```
Comment (robfletcher) [inline gradle.properties]: No more versions in properties

File: gradle/kotlin.gradle [hunk 5]
```diff
@@ -23,19 +23,14 @@ allOpen {
   annotation("com.netflix.spinnaker.KotlinOpen")
 }
 
-configurations.all {
-  resolutionStrategy {
-    eachDependency { details ->
-      if (details.requested.group == "org.jetbrains.kotlin") {
-        details.useVersion kotlinVersion
-      }
-    }
```
Comment (robfletcher) [inline gradle/kotlin.gradle]: Again, not needed because we have the kotlin-bom

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.7-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.3.1-bin.zip
```
Comment (robfletcher) [inline gradle/wrapper/gradle-wrapper.properties]: Gradle 5+ required for bom support

File: keiko-tck/src/main/kotlin/com/netflix/spinnaker/q/QueueTest.kt [hunk 7]
```diff
@@ -261,7 +261,7 @@ abstract class QueueTest<out Q : Queue>(
         }
       }
 
-      it ("retries the message") {
+      it("retries the message") {
```
Comment (cfieber) [inline keiko-tck/src/main/kotlin/com/netflix/spinnaker/q/QueueTest.kt]: `it("retries the message or else it gets the hose again")`

File: keiko-mem/build.gradle [hunk 8]
```diff
@@ -1,9 +1,9 @@
...
 dependencies {
-  compile project(":keiko-core")
-  compile "org.threeten:threeten-extra:1.0"
+  api project(":keiko-core")
+  api "org.threeten:threeten-extra"
 
-  testCompile project(":keiko-tck")
-  testCompile "org.funktionale:funktionale-partials:1.1"
+  testImplementation project(":keiko-tck")
+  testImplementation "org.funktionale:funktionale-partials"
```
Comment (marchello2000) [inline keiko-mem/build.gradle]: this was bumped to 1.2 with the bom, just making sure it's ok

**Comment by marchello2000** [comment]
👍

**Pull request merged**